mat64/dense.go

```diff
@@ -291,10 +293,18 @@ func (m *Dense) Grow(r, c int) Matrix {
 		t.capRows = cr
 		t.capCols = cc
 		// Copy the complete matrix over to the new matrix.
-		// Including elements not currently visible.
-		r, c, m.mat.Rows, m.mat.Cols = m.mat.Rows, m.mat.Cols, m.capRows, m.capCols
-		t.Copy(m)
-		m.mat.Rows, m.mat.Cols = r, c
+		// Including elements not currently visible. Use a temporary structure
```

Comment:
Do we know if tmp escapes? The assumption that it did is probably why I wrote it the way I did.